File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        
        if(root==null)
            return 0;
        
        int left=0;
        int right=0;
        if(root.left!=null)
        {
            left=minDepth(root.left);    
        }
        if(root.right!=null)
        {
            right=minDepth(root.right);
        }
        
        if(left!=0 && right!=0)
            return Math.min(left,right)+1;
        
        if(left!=0)
            return left+1;
        
        return right+1;
        
    }
}

File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MinDepth(TreeNode root) {
        
        
        int min=Int32.MaxValue;
        
        return dfs(root,min);
        
    }
    
    int dfs(TreeNode root,int min)
    {
        if(root==null)
        {
            return 0;
        }
        
        if(root.left==null && root.right==null)
        {
            return 1;
        }
        
        if(root.left!=null)
        {
            min=Math.Min(dfs(root.left,min),min);
        }
        if(root.right!=null)
        {
            min=Math.Min(dfs(root.right,min),min);
        }
        
        return min+1;
    }
}